bentoml/yatai/python_api.py

```diff
@@ -149,3 +164,193 @@ def update_bento_upload_progress(
+                'instance_count': operator_spec.get('instance_count')
+                or config().getint('sagemaker', 'default_instance_count'),
+                'instance_type': operator_spec.get('instance_type')
+                or config().get('sagemaker', 'default_instance_type'),
+                'api_name': operator_spec.get('api_name', ''),
+            }
+        elif operator_value == DeploymentSpec.AWS_LAMBDA:
+            deployment_dict['spec']['aws_lambda_operator_config'] = {
+                'region': operator_spec.get('region')
+                or config().get('aws', 'default_region')
+            }
+            if operator_spec.get('api_name'):
+                deployment_dict['spec']['aws_lambda_operator_config'][
+                    'api_name'
+                ] = operator_spec['api_name']
+        elif operator_value == DeploymentSpec.GCP_FCUNTION:
+            deployment_dict['spec']['gcp_function_operatorConfig'] = {
+                'region': operator_spec.get('region')
+                or config().get('google-cloud', 'default_region')
+            }
+            if operator_spec.get('api_name'):
+                deployment_dict['spec']['gcp_function_operator_config'][
+                    'api_name'
+                ] = operator_spec['api_name']
+        elif operator_value == DeploymentSpec.KUBERNETES:
+            deployment_dict['spec']['kubernetes_operator_config'] = {
+                'kube_namespace': operator_spec.get('kube_namespace', ''),
+                'replicas': operator_spec.get('replicas', 0),
+                'service_name': operator_spec.get('service_name', ''),
+                'service_type': operator_spec.get('service_type', ''),
+            }
+        else:
+            raise BentoMLDeploymentException(
+                'Custom deployment is not supported in the current version of BentoML'
+            )
+
+        return apply_deployment(deployment_dict, yatai_service)
+    except BentoMLDeploymentException as error:
+        return ApplyDeploymentResponse(status=Status.ABORTED(str(error)))
+    except BentoMLException as error:
+        return ApplyDeploymentResponse(status=Status.INTERNAL(str(error)))
+
+
+# TODO update_deployment is not finished.  It will be working on along with cli command
+def update_deployment(deployment_name, namespace, yatai_service=None):
+    raise NotImplemented
+
+
+def apply_deployment(deployment_info, yatai_service=None):
+    if yatai_service is None:
+        from bentoml.yatai import get_yatai_service
+
+        yatai_service = get_yatai_service()
+
+    try:
+        if isinstance(deployment_info, dict):
+            deployment_pb = deployment_dict_to_pb(deployment_info)
+        elif isinstance(deployment_info, str):
+            deployment_pb = deployment_yaml_string_to_pb(deployment_info)
+        else:
+            raise BentoMLDeploymentException(
+                'Currently, we do not handle deployment info type {}'.format(
```

Comment:
'Unexpected argument type, expect deployment_info to be str in yaml format or a dict'